test/e2e/feature/feature.go

```diff
@@ -28,7 +28,10 @@ var (
 	// TODO: document the feature (owning SIG, when to use this feature for a test)
 	APIServerIdentity = framework.WithFeature(framework.ValidFeatures.Add("APIServerIdentity"))
 
-	// TODO: document the feature (owning SIG, when to use this feature for a test)
+	// Owner: sig-lifecycle
+	// This label is used for tests which need the following controllers to be enabled:
+	// - bootstrap-signer-controller
+	// - token-cleaner-controller
```

Comment:
Is it possible to make the above controller enabled by default in kcm? I think these controller is widely used in the industry.

- https://github.com/carlory/kubernetes/blob/e2e-lifecycle/cmd/kubeadm/app/phases/controlplane/manifests.go#L325

I'm not sure whether the following tests depend on a feature label when the cluster is set up with kubeadm without extra configuration.

- https://github.com/kubernetes/kubernetes/blob/f2cfbf44b1fb482671aedbfff820ae2af256a389/test/e2e/lifecycle/bootstrap/bootstrap_signer.go
- https://github.com/kubernetes/kubernetes/blob/f2cfbf44b1fb482671aedbfff820ae2af256a389/test/e2e/lifecycle/bootstrap/bootstrap_token_cleaner.go

Or switch the feature label to `kind`?

- https://github.com/carlory/kubernetes/blob/e2e-lifecycle/test/e2e/feature/feature.go#L200

/cc @pohly @pacoxu @neolit123

Reply:
i think the owner of the bootstrap token feature has always been sig auth.
sig cl was never claimed as the owner of the related controllers bootstrap-signer-controller and token-cleaner-controller, either. sig cl is more of a participating sig as this feature was created mainly because of kubeadm.

maybe sig auth would like to transfer ownership, in which case relevant OWNERS files must be updated.

it terms of enabling these controllers by default, the argument has always been that bs tokens are not a preferred way of joining nodes, as they are insecure.

Reply:
This feature label is only used in the `e2e/lifecycle` directory. `e2e/lifecycle/bootstrap` doesn't contains a OWNERS file. All tests under this directory is owned by sig-lifecycle, not sig-auth. That is why I assume the owner of this test and feature label is sig-cluster-lifecycle.

```
approvers:
  - neolit123
  - SataQiu
  - pacoxu
reviewers:
  - neolit123
  - SataQiu
  - pacoxu
labels:
  - sig/cluster-lifecycle
...
```

Should I add a new OWNERS file to `e2e/lifecycle/bootstrap` or move all tests to `e2e/auth`?

Reply:
/cc @liggitt

Reply:
> This feature label is only used in the e2e/lifecycle directory. e2e/lifecycle/bootstrap doesn't contains a OWNERS file. All tests under this directory is owned by sig-lifecycle, not sig-auth. That is why I assume the owner of this test and feature label is sig-cluster-lifecycle.

ok, in that case this PR, LGTM

Reply:
/lgtm

Reply:
> Is it possible to make the above controller enabled by default in kcm? I think these controller is widely used in the industry.

No, bootstrap tokens must be opted into by the cluster administrator. They are not a default authentication method. It was created specifically for `kubeadm`, though other deployments can make use of it. It is a non-default configuration from the perspective of kube-apiserver and kube-controller-manager.

> terms of enabling these controllers by default, the argument has always been that bs tokens are not a preferred way of joining nodes, as they are insecure.

+1 to this